code/modules/research/weaponsanalyzer.dm

```diff
@@ -72,24 +63,31 @@
 		process = TRUE
 		update_icon()
 	else if(I)
+		check_swap(user, I)
 		item = I
 		H.drop_from_inventory(I)
 		I.forceMove(src)
 		update_icon()
 
-/obj/machinery/weapons_analyzer/attack_hand(mob/user as mob)
+/obj/machinery/weapons_analyzer/attack_hand(mob/user)
 	user.set_machine(src)
 	ui_interact(user)
 
 /obj/machinery/weapons_analyzer/proc/reset()
 	process = FALSE
 	update_icon()
 
-/obj/machinery/weapons_analyzer/proc/check_gun(var/mob/user)
+/obj/machinery/weapons_analyzer/proc/check_swap(var/mob/user, var/obj/I)
 	if(item)
-		to_chat(user, span("warning", "\The [src] already has \the [item] mounted. Remove it first."))
-		return FALSE
-	return TRUE
+		to_chat(user, SPAN_NOTICE("You swap \the [item] out for \the [I]."))
+		if(istype(item, /obj/item/device/laser_assembly))
+			var/obj/item/device/laser_assembly/A = item
+			A.ready_to_craft = FALSE
+			A.analyzer = null
+		item = null
```

Comment:
You need to move this down below line 90. Because you null it here and then line 89 will runtime. Line 90 wouldn't get executed, but even if it was, then the null ref will do nothing. 

Another thing, I assume you are swapping item, so then this should be `item = I` below line 90.